pkg/api/context.go

```diff
@@ -26,6 +27,9 @@ import (
 // Context carries values across API boundaries.
 type Context interface {
 	Value(key interface{}) interface{}
+	Deadline() (deadline time.Time, ok bool)
```

Comment:
Actually no, b/c in order for the api.Context to be passed through the storage layer they are needed.